### DISCUSSION TITLE
Firebase image upload not working on emulator with React Native

### DISCUSSION THREAD

**Discussion by ozgurcrs** [invertase/react-native-firebase#4673] [Q&A]
Hi guys,
I am trying to make an upload image with Firebase storage.
While uploading, it works on an Android device but I get an error on the emulator. It is important for me that the emulator also works. Can you help? 

Error and main manifest
`<manifest xmlns:android="http://schemas.android.com/apk/res/android"
  package="com.facedetection">

    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
    <uses-permission android:name="android.permission.CAMERA"/>
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.INTERNET"/>
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />

    <application
      android:name=".MainApplication"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:requestLegacyExternalStorage="true"
      android:roundIcon="@mipmap/ic_launcher_round"
      android:allowBackup="false"
      android:theme="@style/AppTheme">
      <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode"
        android:launchMode="singleTask"
        android:windowSoftInputMode="adjustResize">
        <intent-filter>
            <action android:name="android.intent.action.MAIN" />
            <category android:name="android.intent.category.LAUNCHER" />
        </intent-filter>
      </activity>
      <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
    </application>

</manifest>
`

Error and debug manifest
`<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools">

    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
    <uses-permission android:name="android.permission.CAMERA"/>
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.INTERNET"/>
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />

    

    <application android:usesCleartextTraffic="true"
     tools:targetApi="28" 
     android:requestLegacyExternalStorage="true"
     tools:ignore="GoogleAppIndexingWarning" />
</manifest>
`

Last one  My upload function

` saveItem = async () => {
    const imageInfo = this.state.setImageSource;
    try {
      console.log(imageInfo.fileName);
      console.log(imageInfo.uri);
      await storage().ref(imageInfo.fileName).putFile(imageInfo.uri);
          console.log("Yüklendi");
          Alert.alert("Başarılı");
    }
    catch (e) {
      console.log(e)
    }
  }`

**Comment by mikehardy** [comment]
Hi there!
- Please keep it textual, images are not very useful to me as I am on small devices travelling when I take time to scan issues normally

- You are having an issue but yo opened a discussion, which has none of the information we would need to troubleshoot

Please open a fresh issue, with a reproduction of the problem that fits in a clean / small App.js we can drop in a sample project (like that from https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh), filling in template details

Thank you

**Reply by ozgurcrs** [reply]
I edited thanks

**Reply by mikehardy** [reply]
Editing is not following instructions, sorry

> Please open a fresh issue, with a reproduction of the problem that fits in a clean / small App.js we can drop in a sample project (like that from https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh), filling in template details
